tgui/packages/tgui/interfaces/RCD.js

```diff
@@ -0,0 +1,258 @@
+import { useBackend } from "../backend";
+import { Button, Section, LabeledList, Box, ProgressBar, Flex, Tabs, Icon } from "../components";
+import { Window } from "../layouts";
+import { ComplexModal, modalOpen } from "./common/ComplexModal";
+import { AccessList } from './common/AccessList';
+import { Fragment } from "inferno";
+
+export const RCD = () => {
+  return (
+    <Window resizable>
+      <ComplexModal />
+      <Window.Content display="flex" className="Layout__content--flexColumn">
+        <MatterReadout />
+        <ConstructionType />
+        <AirlockSettings />
+        <TypesAndAccess />
+      </Window.Content>
+    </Window>
+  );
+};
+
+const MatterReadout = (props, context) => {
+  const { data } = useBackend(context);
+  const {
+    matter,
+    max_matter,
+  } = data;
+  const good_matter = max_matter * 0.70;
+  const average_matter = max_matter * 0.25;
+  return (
+    <Section
+      title="Matter Storage"
+      flexBasis="content">
+      <ProgressBar
+        ranges={{
+          good: [good_matter, Infinity],
+          average: [average_matter, good_matter],
+          bad: [-Infinity, average_matter],
+        }}
+        value={matter}
+        maxValue={max_matter}>
+        <Box textAlign="center">
+          {matter + " / " + max_matter + " units"}
+        </Box>
+      </ProgressBar>
+    </Section>
+  );
+};
+
+const ConstructionType = () => {
+  return (
+    <Section
+      title="Construction Type"
+      flexBasis="content">
+      <Flex>
+        <Flex.Item>
+          <ConstructionTypeCheckbox mode_type="Floors and Walls" />
+          <ConstructionTypeCheckbox mode_type="Airlocks" />
+          <ConstructionTypeCheckbox mode_type="Windows" />
+          <ConstructionTypeCheckbox mode_type="Deconstruction" />
+        </Flex.Item>
+      </Flex>
+    </Section>
+  );
+};
+
+const ConstructionTypeCheckbox = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { mode_type } = props;
+  return (
+    <Button.Checkbox
+      content={mode_type}
+      checked={data.mode === mode_type ? 1 : 0}
```

Comment:
Add mode as a var on line 69, since you already did mode_type

Reply:
Do you mean `const { mode } = data;`?

Reply:
```js
const {
  mode,
  mode_type,
} = data;
```

Reply:
Right but `mode_type` is from props, so don't they have to be separate? i.e
```js
const { mode_type } = props;
const { mode } = data;
```
?

Reply:
Yes, apologies